src/MySqlConnector/Protocol/Payloads/OkPayload.cs

```diff
@@ -79,7 +75,28 @@ internal sealed class OkPayload
 						switch (kind)
 						{
 							case SessionTrackKind.Schema:
-								newSchema = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								context.Database = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								break;
+
+							case SessionTrackKind.SystemVariables:
+								var systemVariableOffset = reader.Offset + dataLength;
+								do
+								{
+									var variableSv = Encoding.ASCII.GetString(reader.ReadLengthEncodedByteString());
+									var lenSv = reader.ReadLengthEncodedIntegerOrNull();
+									var valueSv = lenSv == -1
+										? null
+										: Encoding.ASCII.GetString(reader.ReadByteString(lenSv));
+									switch (variableSv)
+									{
+										case "character_set_client":
```

Comment:
This is just one of the three possible connection-level character set variables, including `character_set_connection` and `character_set_results`. It seems prudent to verify that all three have the expected value (since the server will send them to us). Updated here: https://github.com/mysql-net/MySqlConnector/pull/1497/commits/3f50c14f734b23025d98f0d090416cec2a1940cf.